modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/NavigationPanel.js

```diff
@@ -58,8 +68,34 @@ export default function NavigationPanel({items, selectedItemId}) {
 
 						<ClayTreeView.Group items={item.children}>
 							{(item) => {
+								let transformedActions;
+
+								if (item?.actions?.length) {
+									transformedActions = propsTransformer({
+										items: normalizeDropdownItems(
+											item.actions
+										),
+									});
+								}
+
 								return (
 									<ClayTreeView.Item
+										actions={
+											item?.actions?.length && (
+												<ClayDropDownWithItems
+													items={
+														transformedActions.items
```

Comment:
Can we simply inline this? 
If we inline we avoid checking two times for `item?.actions?.length` and creating `transformedActions`.

Something like this (after moving propsTransformer inside of  normalizeDropdownItems)

```jsx
items={normalizeDropdownItems(item.actions)}
```